api/vpc/v1alpha2/vpc_types.go

```diff
@@ -329,7 +338,13 @@ func (vpc *VPC) Validate(ctx context.Context, kube client.Reader, fabricCfg *met
 			if !ipNet.Contains(ip) {
 				return nil, errors.Errorf("subnet %s: dhcp range start %s is not in the subnet", subnetName, subnetCfg.DHCP.Range.Start)
 			}
+			if ip := net.ParseIP(subnetCfg.DHCP.DNSServer); ip == nil && len(subnetCfg.DHCP.DNSServer) > 0 {
+				return nil, errors.Errorf("subnet %s: dns address is not a valid IP", subnetName)
+			}
 
+			if ip := net.ParseIP(subnetCfg.DHCP.TimeServer); ip == nil {
```

Comment:
I guess this check should be used only if TimeServer len is > 0, additionally it'll only allow IP address while your error message is mentioning fqdn

Reply:
length check not required parseip will fail for zero length strings as well

Reply:
if it's an empty string there should be no error message as it's an optional parameter, but you'll get one now as ParseIP will return nil on empty string. Additionally, your error message shouldn't mention FQDN as you aren't allowing it here.